lib/spack/spack/test/compilers.py

```diff
@@ -81,3 +81,145 @@ def test_compiler_flags_from_config_are_grouped():
 
     compiler = compilers.compiler_from_config_entry(compiler_entry)
     assert any(x == '-foo-flag foo-val' for x in compiler.flags['cflags'])
+
+
+# Test behavior of flags and UnsupportedCompilerFlag.
+
+# Utility function to test most flags.
+default_compiler_entry = {
+    'spec': 'clang@2.0.0-apple',
+    'operating_system': 'foo-os',
+    'paths': {
+        'cc': 'cc-path',
+        'cxx': 'cxx-path',
+        'fc': None,
+        'f77': None
+    },
+    'flags': {},
+    'modules': None
+}
+
+
+# Fake up a generic compiler where everything is defaulted.
+class GenericCompiler(Compiler):
+    def __init__(self):
+        super(GenericCompiler, self).__init__(
+            "badcompiler@1.0.0",
+            default_compiler_entry['operating_system'],
+            None,
+            [default_compiler_entry['paths']['cc'],
+             default_compiler_entry['paths']['cxx'],
+             default_compiler_entry['paths']['fc'],
+             default_compiler_entry['paths']['f77']])
+
+    @property
+    def name(self):
+        return "genericcompiler"
+
+    @property
+    def version(self):
+        return "1.0.0"
+
+
+# Utility function to verify that the expected exception is thrown for
+# an unsupported flag.
+def unsupported_flag_test(flag, compiler_entry=default_compiler_entry):
+    compiler = None
+    if compiler_entry is None:
+        compiler = GenericCompiler()
+    else:
+        # flake8 note: we *are* going to use compiler, but in an eval
```

Comment:
Here is the block at issue. Note that the problem is not the long line, but that the long line is mandated by the need to deactivate the unused variable warning (`eval` does not qualify as *using* a variable, unfortunately). As a result, the long line warning now needs to be deactivated also.